chain/chain/src/store.rs

```diff
@@ -2465,6 +2466,103 @@ impl<'a> ChainStoreUpdate<'a> {
         Ok(())
     }
 
+    /// Only used in mock network
+    /// Copy the necessary chain state related to `block_hash` from `source_store` to the current
+    /// store.
+    pub fn copy_chain_state_as_of_block(
```

Comment:
why do we need to update all the caches?

Reply:
It's not for updating the cache, eventually these changes to will be commit to the database